crates/wasi-http/Cargo.toml

```diff
@@ -27,6 +27,7 @@ http-body = { workspace = true }
 http-body-util = { workspace = true }
 tracing = { workspace = true }
 wasmtime-wasi = { workspace = true }
+wasmtime-wasi-io = { workspace = true }
```

Comment:
As a proof-of-concept would it be possible to not add the dependency here? Or is it required that those using `wasmtime-wasi` have to also depend on `wasmtime-wasi-io`?

Reply:
Yes, its possible to not add the dependency and `use` everything through wasmtime-wasi. I'm happy to convert it over to that if you think that is valuable, I could go either way.

Reply:
Yeah if you wouldn't mind doing that I think it'd be good to have an example and basically a test for that. In Spin for example I'd hope to ideally only need to refactor a bit and not have to explicitly pick up `wasmtime-wasi-io` (otherwise we might need to extend docs more to mention it)

Reply:
Done - hitting merge.